lib/sly-completion.el

```diff
@@ -278,53 +282,77 @@ ANNOTATION) describing each completion possibility."
-          :company-require-match 'never
-          :company-match
-          (lambda (obj)
-            (get-text-property 0 'sly-completion-chunks obj))
-          :company-location
-          (lambda (obj)
-            (save-window-excursion
-              (let* ((buffer (sly-edit-definition
-                              (substring-no-properties obj))))
-                (when (buffer-live-p buffer) ; on the safe side
-                  (cons buffer (with-current-buffer buffer
-                                 (point)))))))
-          :company-prefix-length
-          (and (sly--completion-inside-string-or-comment-p) 0))))
+    (append
+     (list beg end
+           (sly--completion-function-wrapper fn)
+           :annotation-function #'sly-completion-annotation
+           :exit-function (lambda (obj _status)
+                            (let ((suggestion
+                                   (get-text-property 0 'sly--suggestion
+                                                      obj)))
+                              (when suggestion
+                                (delete-region (- (point) (length obj)) (point))
+                                (insert suggestion))))
+           :company-docsig
+           (lambda (obj)
+             (when (sit-for 0.1)
+               (sly--responsive-eval (arglist `(slynk:operator-arglist
+                                                ,(substring-no-properties obj)
+                                                ,(sly-current-package)))
+                                     (or (and arglist
+                                              (sly-autodoc--fontify arglist))
+                                         "no autodoc information"))))
+           :company-no-cache t
+           :company-doc-buffer
+           (lambda (obj)
+             (when (sit-for 0.1)
+               (sly--responsive-eval (doc `(slynk:describe-symbol
+                                            ,(substring-no-properties obj)))
+                                     (when doc
+                                       (with-current-buffer (get-buffer-create " *sly-completion doc*")
+                                         (erase-buffer)
+                                         (insert doc)
+                                         (current-buffer))))))
+           :company-require-match 'never
+           :company-match
+           (lambda (obj)
+             (get-text-property 0 'sly-completion-chunks obj))
+           :company-location
+           (lambda (obj)
+             (save-window-excursion
+               (let* ((buffer (sly-edit-definition
+                               (substring-no-properties obj))))
+                 (when (buffer-live-p buffer) ; on the safe side
+                   (cons buffer (with-current-buffer buffer
+                                  (point)))))))
+           :company-prefix-length
+           (and (sly--completion-inside-string-or-comment-p) 0))
+     (sly--company-kind-perperty))))
```

Comment:
`perperty` ? 😂 

Leave the previous version, or use:

```
`(,beg ,end
  ...lots of stuff  (but much of it needs commas since it's not evaluated)
  ,@(when shubidu
      `(:company-kind (lambda (comp) ...))))
```

Reply:
Ok, back to previous version. I don't really like the `append` but neither the quoted list. :)

Reply:
> Ok, back to previous version. I don't really like the append but neither the quoted list. :)

backquoted.  

Well, you can use `list*`

Reply:
> > Ok, back to previous version. I don't really like the append but neither the quoted list. :)
> 
> backquoted.

I meant not per-se. Backquoted list is OK. But in this case there would be too many unquotes necessary.

> 
> Well, you can use `list*`

Good point. But I'd leave it as is.

Should I now look at the GNU ChangeLog or will you?